src/script/interpreter.cpp

```diff
@@ -1267,12 +1385,62 @@ bool EvalScript(std::vector<std::vector<unsigned char> >& stack, const CScript&
                     if (stack.size() < 2) {
                         return set_error(serror, SCRIPT_ERR_INVALID_STACK_OPERATION);
                     }
-                    if (!checker.CheckUnvaultTriggerOutputs(
-                            recovery_spk_hash, spend_delay, fRequireMinimal)) {
+
+                    CScript unvault_output_spk;
+
+                    if (!checker.CheckUnvaultTriggerOutputsCommon(
+                            recovery_spk_hash,
+                            spend_delay,
+                            fRequireMinimal,
+                            unvault_output_spk)) {
                         return set_error(serror, SCRIPT_ERR_UNVAULT_MISMATCH);
                     }
+                    assert(unvault_output_spk.size() > 0);
+
+                    // Determine the type of the proposed OP_UNVAULT script.
+                    // If it lives behind a scripthash, we will have to construct
+                    // a specific expected scriptPubKey and compare to that. Otherwise
+                    // we'll analyze its prefix as a bare script.
+                    int opuv_witversion;
+                    valtype opuv_witprogram;
+                    const bool is_unvault_output_wit =
+                        unvault_output_spk.IsWitnessProgram(opuv_witversion, opuv_witprogram);
+
+                    if (!is_unvault_output_wit) {
+                        // Validate the OP_UNVAULT output as a bare script.
+                        if (!CheckUnvaultTriggerOutputsBare(
+                                unvault_output_spk,
+                                recovery_spk_hash, spend_delay, fRequireMinimal)) {
+                            return set_error(serror, SCRIPT_ERR_UNVAULT_MISMATCH);
+                        }
```

Comment:
In a9db7392bf9d4d69d5a7a91db32dda7bd50479d8 "consensus: allow OP_UNVAULT outputs to be behind scripthash"

Is it useful to allow OP_UNVAULT as a bare script? ISTM it would be simpler to just require that the the unvault script must be one of P2WSH or P2TR. It's not clear to me that it's actually useful to allow it as a bare script, especially since that requires further policy changes to allow such transactions to even be relayed.

Reply:
Yeah, this is a good point. I think I'll remove this.

Reply:
Removed. Vault opcodes now only exist as OP_SUCCESSx overrides in tapscript.